extensions/sql-migration/src/wizard/createMigrationControllerDialog.ts

```diff
@@ -0,0 +1,485 @@
+	private migrationControllerResourceGroupDropdown!: azdata.DropDownComponent;
+	private migrationControllerRegionDropdown!: azdata.DropDownComponent;
+	private migrationControllerNameText!: azdata.InputBoxComponent;
+	private _formSubmitButton!: azdata.ButtonComponent;
+
+	private _statusLoadingComponent!: azdata.LoadingComponent;
+	private migrationControllerAuthKeyTable!: azdata.DeclarativeTableComponent;
+	private _connectionStatus!: azdata.TextComponent;
+	private _copyKey1Button!: azdata.ButtonComponent;
+	private _copyKey2Button!: azdata.ButtonComponent;
+	private _setupContainer!: azdata.FlexContainer;
+
+	private _dialogObject!: azdata.window.Dialog;
+	private _view!: azdata.ModelView;
+	private _subscriptionMap: Map<string, Subscription> = new Map();
+
+	constructor(private migrationStateModel: MigrationStateModel, private irPage: IntergrationRuntimePage) {
+		this._dialogObject = azdata.window.createModelViewDialog(constants.IR_PAGE_TITLE, 'MigrationControllerDialog', 'wide');
+	}
+
+	initialize() {
+		let tab = azdata.window.createTab('');
+		this._dialogObject.registerCloseValidator(async () => {
+			return true;
+		});
+		tab.registerContent((view: azdata.ModelView) => {
+			this._view = view;
+
+			this._formSubmitButton = view.modelBuilder.button().withProps({
+				label: constants.SUBMIT,
+				width: '80px'
+			}).component();
+
+			this._formSubmitButton.onDidClick(async (e) => {
+				this._statusLoadingComponent.loading = true;
+				this._formSubmitButton.enabled = false;
+
+				const subscription = this._subscriptionMap.get((this.migrationControllerSubscriptionDropdown.value as azdata.CategoryValue).name)!;
+				const resourceGroup = (this.migrationControllerResourceGroupDropdown.value as azdata.CategoryValue).name;
+				const region = (this.migrationControllerRegionDropdown.value as azdata.CategoryValue).name;
+				const controllerName = this.migrationControllerNameText.value;
+
+				const formValidationErrors = this.validateCreateControllerForm(subscription, resourceGroup, region, controllerName);
+
+				if (formValidationErrors.length > 0) {
+					this.setDialogMessage(formValidationErrors);
+					this._statusLoadingComponent.loading = false;
+					this._formSubmitButton.enabled = true;
+					return;
+				}
+
+				const createdController = await createMigrationController(this.migrationStateModel.azureAccount, subscription, resourceGroup, region, controllerName!);
+				if (createdController.error) {
+					this.setDialogMessage(`${createdController.error.code} : ${createdController.error.message}`);
+					this._statusLoadingComponent.loading = false;
+					this._formSubmitButton.enabled = true;
+					return;
+				}
+
+				this._dialogObject.message = {
+					text: ''
+				};
+				this.migrationStateModel.migrationController = createdController;
+				await this.refreshAuthTable();
+				await this.refreshStatus();
+				this._setupContainer.display = 'inline';
+				this._formSubmitButton.enabled = false;
```

Comment:
What happens if an exception is thrown during execution of this? Generally you should be wrapping this stuff in a try/catch and then ensuring that buttons/loading status' are reset correctly if that happens